feat: survey access controls

zlwaterfield opened the pull request:
## Changes

Adding survey access controls.

Note - this doesn't yet handle the connection to the linked flags. But should work for everything else. 

Loom walkthrough: https://www.loom.com/share/0ab8d4c7360c442994a3f7c73772c052?sid=995a50da-172c-41a0-8e17-7d0c99edbbae

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Optional, but helpful for our content team! -->
<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->

zlwaterfield committed:
Initial commit for survey access controls

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

zlwaterfield committed:
survey button changes

zlwaterfield committed:
Finish up survey access controls

zlwaterfield committed:
Update LaunchSurveyButton.tsx

a-lider commented on frontend/src/scenes/surveys/SurveySettings.tsx:
```diff
@@ -126,46 +132,68 @@ export function SurveySettings({ isModal = false }: Props): JSX.Element {
+                {({ disabledReason }) => {
+                    // The disabledReason is set if the user doesn't have access to the survey resource
+                    if (disabledReason) {
+                        return null
```
We should probably show the user why they don't see any settings. For example, a LemonBanner with the disabled reason.

zlwaterfield committed:
Merge remote-tracking branch 'origin/master' into zach/surveys-access-controls

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

zlwaterfield committed:
Merge remote-tracking branch 'origin/master' into zach/surveys-access-controls

zlwaterfield committed:
Add user_access_level to Survey mocks and update access control context

Added the user_access_level property to Survey mock objects and test data to support access control features. Updated surveyLogic to include access control resource context for surveys, and fixed type casting in SidePanelAccessControl for resource type. Removed redundant SIDE_PANEL_CONTEXT_KEY logic from surveyLogic.

zlwaterfield committed:
Update test_survey.py

github-actions[bot] committed:
Update query snapshots

a-lider reviewed (approved):
Tested locally, everything works great!
Just two comments:

1 - Didn't immediately find access control in the side panel for surveys. It shows up on an individual survey, but not on the main surveys page.  I think it's worth adding if this is a reference PR. 

2 - Noticed that the survey preview is flickering in the Settings tab after the page loads. Looks like it's rerendering. This doesn't happen on master. 
![2025-10-08 14 39 15](https://github.com/user-attachments/assets/b62cd62e-9401-415b-a577-97632eebc223)
